server/sonar-server/src/test/java/org/sonar/server/computation/issue/RuleCacheLoaderTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.issue;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.core.persistence.DbTester;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.rule.db.RuleDao;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+public class RuleCacheLoaderTest {
+
+  @Rule
+  public DbTester dbTester = new DbTester();
+
+  @Test
+  public void load_by_key() throws Exception {
+    dbTester.prepareDbUnit(getClass(), "shared.xml");
+    DbClient dbClient = new DbClient(dbTester.database(), dbTester.myBatis(), new RuleDao());
+    RuleCacheLoader loader = new RuleCacheLoader(dbClient);
+
+    assertThat(loader.load(RuleKey.of("squid", "R001")).getName()).isEqualTo("Rule One");
+    assertThat(loader.load(RuleKey.of("squid", "MISSING"))).isNull();
+  }
+
+  @Test
```

Comment:
two solutions: use @Test(expected:UnsupportedOperationException) or use https://github.com/Codearte/catch-exception to keep the arrange/act/assert test pattern